# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/linux/eth-naming.patch
# Copyright (C) 2006 - 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

Ever chaning wlan^Wwifi^Weth interface names do not only annoy users, but
also prevent reusing configuration from one machine to another.

Sanitze naming to alway be wlan* regardless of who wrote the driver.

  - Rene Rebe <rene@exactcode.de>

--- linux-4.12/drivers/net/wireless/marvell/mwifiex/main.c.vanilla	2017-07-06 14:02:56.949536458 +0200
+++ linux-4.12/drivers/net/wireless/marvell/mwifiex/main.c	2017-07-06 14:03:08.189536198 +0200
@@ -594,7 +594,7 @@
 
 	rtnl_lock();
 	/* Create station interface by default */
-	wdev = mwifiex_add_virtual_intf(adapter->wiphy, "mlan%d", NET_NAME_ENUM,
+	wdev = mwifiex_add_virtual_intf(adapter->wiphy, "wlan%d", NET_NAME_ENUM,
 					NL80211_IFTYPE_STATION, NULL);
 	if (IS_ERR(wdev)) {
 		mwifiex_dbg(adapter, ERROR,
--- linux-6.8/drivers/net/wireless/intersil/orinoco/main.c.vanilla	2024-05-05 19:52:17.931742115 +0200
+++ linux-6.8/drivers/net/wireless/intersil/orinoco/main.c	2024-05-05 19:52:33.644743371 +0200
@@ -2274,6 +2274,7 @@
 	dev->max_mtu = ORINOCO_MAX_MTU;
 
 	SET_NETDEV_DEV(dev, priv->dev);
+	strcpy(dev->name, "wlan%d");
 	ret = register_netdev(dev);
 	if (ret)
 		goto fail;
